src/accelerate/commands/launch.py

```diff
@@ -692,16 +696,26 @@ def launch_command(args):
             else:
                 args.mixed_precision = defaults.mixed_precision
     else:
+        warned = False
         if args.num_processes is None:
+            logger.warn("`--num_processes` was not set, using a value of `1`.")
+            warned = True
             args.num_processes = 1
         if args.num_machines is None:
+            warned = True
+            logger.warn("`--num_machines` was not set, using a value of `1`.")
             args.num_machines = 1
         if args.mixed_precision is None:
+            warned = True
+            logger.warn("`--mixed_precision` was not set, using a value of `'no'`.")
             args.mixed_precision = "no"
         if not hasattr(args, "use_cpu"):
-            args.use_cpu = torch.cuda.is_available()
+            args.use_cpu = args.cpu
+        if warned:
+            logger.warn("To avoid these warnings pass in values for each of the problematic parameters")
```

Comment:
```suggestion
            logger.warn("To avoid these warnings pass in values for each of the problematic parameters or run `accelerate config`.")
```